### DISCUSSION TITLE
Please help me confirm how to configure SkyWalking for cross-cluster tracing.

### DISCUSSION THREAD

**Discussion by Hoverhuang-er** [apache/skywalking#11423] [General]
Hi 
Can you please tell me how to configure SkyWalking for cross-cluster calls in Kubernetes? Currently, it is known that calling services in cluster B from cluster A will cause the call chain to be lost. Does SkyWalking support cross-cluster calls and querying in OAP? At the same time, it is also necessary to confirm whether SkyWalking's data collect protocol also supports traces data for cross-cluster calls.

**Comment by wu-sheng** [comment]
What do you mean cross cluster? OAP cross clusters or monitored services cross clusters.
Could you share how things deployed?

**Reply by Hoverhuang-er** [reply]
In this case, the monitored microservices are deployed in multiple Kubernetes clusters. Services call each other through NodeIP or PodIP. SkyWalking agents are injected into each service deployed in different Kubernetes clusters. Can SkyWalking support microservices tracing for cross-cluster calls?

**Reply by wu-sheng** [reply]
Are these two cluster services reporting to a unified OAP cluster?

**Reply by Hoverhuang-er** [reply]
Yes. Reporting to unified OAP endpoint

**Reply by wu-sheng** [reply]
You could use this, https://github.com/apache/skywalking-java/blob/017bc20908536c7b16660e2a2452b0478a48f2b6/apm-sniffer/config/agent.config#L27

And you can see explanation from the codes, https://github.com/apache/skywalking-java/blob/017bc20908536c7b16660e2a2452b0478a48f2b6/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java#L75

This is not a popular use case, we don't document it yet.

**Reply by wu-sheng** [reply]
And, I believe this only exists in Java agent, AFAIK.
